src/MassiveAction.php

```diff
@@ -795,15 +795,17 @@ public static function getAllMassiveActions($item, $is_deleted = false, ?CommonD
                     $actions[$self_pref . 'delete'] = _x('button', 'Put in trashbin');
                 }
             } else if ($canpurge) {
-                if ($item instanceof CommonDBRelation) {
-                    $actions[$self_pref . 'purge'] = _x('button', 'Delete permanently the relation with selected elements');
-                } else {
+                if (!$item instanceof CommonDBRelation) {
                     $actions[$self_pref . 'purge'] = _x('button', 'Delete permanently');
                 }
                 if ($item instanceof CommonDropdown) {
                     $actions[$self_pref . 'purge_but_item_linked']
                      = _x('button', 'Delete permanently even if linked items');
                 }
+            } elseif ($candelete) {
+                if ($item instanceof CommonDBRelation) {
+                    $actions[$self_pref . 'delete'] = _x('button', 'Delete permanently the relation with selected elements');
+                }
```

Comment:
Purging with only the deletion right would set a precedent in GLPI.

If I take the example of `Item_OperatingSystem` which is also a `CommonDBRelation`, the difference (in the code) is that `Item_OperatingSystem` does not depend on the` $rightname = "operatingsystem"`. 

It even specifies to `CommonDBRelation` not to check the rights on the `OperatingSystem` item.

`public static $checkItem_2_Rights = self::DONT_CHECK_ITEM_RIGHTS;`

This simple patch should be sufficient.

```patch
diff --git a/src/Domain_Item.php b/src/Domain_Item.php
index e349b077b6..c1ae17bcf1 100644
--- a/src/Domain_Item.php
+++ b/src/Domain_Item.php
@@ -42,7 +42,7 @@ class Domain_Item extends CommonDBRelation
     public static $itemtype_2 = 'itemtype';
     public static $items_id_2 = 'items_id';
 
-    public static $rightname = 'domain';
+    public static $checkItem_1_Rights = self::DONT_CHECK_ITEM_RIGHTS;
 
     public static function getTypeName($nb = 0)
     {
```

This allows the `purge` of the relation without granting the purge right on `Domain`.

`CommonDBRelation` will verify that the user has the right to `update` the main element, in this case, `Computer`, to allow the purge of the relation.